extension/api.js

```diff
@@ -43,25 +55,51 @@
 	})();
 
 	window.gitHubNotifCount = function (callback) {
-		var tmp = document.createElement('div');
+		var token = GitHubNotify.settings.get('oauthToken');
+		var opts = {
+				Authorization: 'token ' + token
+		};
+		var participating = GitHubNotify.settings.get('useParticipatingCount')
+			? '?participating=true'
+			: '';
+		var url = GitHubNotify.settings.get('rootUrl');
 
-		xhr('GET', GitHubNotify.settings.get('notificationUrl'), function (data, status) {
+		if (!token) {
+			callback(-4);
+			return;
+		}
+		if (/github.com\/?/.test(url)) {
+			url = 'https://api.github.com/notifications';
+		} else {
+			url += 'api/v3/notifications';
+		}
+		url += participating;
+
+		xhr('GET', url, opts, function (data, status, response) {
+			var interval = Number(response.getResponseHeader('X-Poll-Interval'));
 			if (status >= 400) {
-				callback(-1);
+				callback(-1, interval);
+				return;
+			}
+
+			if (status === 304) {
+				callback(-3, interval);
```

Comment:
These arguments (-1, -3, -4) look a bit too cryptic, why not name them to make this more readable?